### PR TITLE
Improve build uploading

### PR DESCRIPTION
**Pull request by janisozaur**
There have been changes to how incoming builds are getting classified
on openrct2.org side, see
https://github.com/Limetric/OpenRCT2.org/blob/e5b738f3dadcc5a3b78e8dfd434756ff31eaa1d3/src/misc/releaseAsset.js#L94-L116,
rendering the 'flavour' mechanism obsolete. Additionally, AppImage
should now be accepted as well.

### CODE REVIEW

**Comment by tupaschoal** [comment]
Is anything holding this back?

**Comment by janisozaur** [comment]
commitlint was the issue. Even after updating the action in #19452 a false positive was still getting reported. I modified the the commit message to remove offending item (See https://github.com/conventional-changelog/commitlint/issues/896 - it was the `#L96` anchor in the link) and manually added it back in the squash commit message through github

**Comment by janisozaur** [comment]
I've temporarily removed requirement status of AppImage builds, as it is getting renamed here. Will restore it once it's merged

**Review by ZehMatt — APPROVED** [review]

**Pull request merged**
